Extend/Payment_Gateway.php

```diff
@@ -187,92 +202,177 @@ public function makeTransaction($dAmount, $oOrder)
         $oTransaction = $oPaymentMethod->getTransaction();
         $oTransaction->setRedirect($oRedirect);
 
-        $config = \OxidEsales\Eshop\Core\Registry::getConfig();
-        $currency = $config->getActShopCurrencyObject();
-        $oTransaction->setAmount(new Amount($dAmount, $currency->name));
+        $oShopconfig = $this->getConfig();
+        $oCurrency = $oShopconfig->getActShopCurrencyObject();
+        $oTransaction->setAmount(new Amount($dAmount, $oCurrency->name));
 
-        $basket = $oSession->getBasket();
-        $user = $basket->getBasketUser();
+        $oBasket = $oSession->getBasket();
+        $oUser = $oBasket->getBasketUser();
 
         $oTransaction->setOrderDetail(sprintf(
             '%s %s %s',
             $oOrder->oxorder__oxbillemail->value,
-            $user->oxuser__oxfname->value,
-            $user->oxuser__oxlname->value
+            $oUser->oxuser__oxfname->value,
+            $oUser->oxuser__oxlname->value
         ));
 
-        $payment = oxNew(\OxidEsales\Eshop\Application\Model\Payment::class);
-        $payment->load(self::NAME);
+        $sPaymentId = $oBasket->getPaymentId();
+        $oPayment = oxNew(Payment::class);
+        $oPayment->load($sPaymentId);
 
-        if ($payment->oxpayments__wdoxidee_additional_info->value) {
-            $oViewConf = oxNew('oxViewConfig');
-            $sIp = $oViewConf->getRemoteAddress();
-            $oTransaction->setIpAddress($sIp);
-            $oTransaction->setConsumerId($user->oxuser__oxid->value);
-            $oTransaction->setOrderNumber($oOrder->oxorder__oxid->value);
+        if ($oPayment->oxpayments__wdoxidee_additional_info->value) {
+            $this->_addAdditionalInfo($oTransaction, $oOrder, $oUser, $oPayment);
         }
 
-        if ($payment->oxpayments__wdoxidee_descriptor->value) {
+        if ($oPayment->oxpayments__wdoxidee_descriptor->value) {
             $descriptor = self::getDescriptor($oOrder->oxorder__oxid->value);
             $oTransaction->setDescriptor($descriptor);
         }
 
-        $address = new \Wirecard\PaymentSdk\Entity\Address(
-            self::getCountryCode($user->oxuser__oxcountryid),
-            $user->oxuser__oxcity->value,
-            $user->oxuser__oxstreet->value
-        );
-        $address->setPostalCode($user->oxuser__oxzip->value);
+        if ($oPayment->oxpayments__wdoxidee_basket->value) { //add basket data
```

Comment:
I would prefer to remove the comment as it is not providing any additional info (add basket data called on the next line).